src/tests/test_targets.py

```diff
@@ -43,3 +45,27 @@ def test_delete_clusters(self, api_client: InventoryClient, cluster_configuratio
             cluster.delete()
 
         log.info(f"Successfully deleted {len(clusters)} clusters")
+
+    @JunitTestSuite()
+    def test_destroy_terraform(
+        self, api_client: InventoryClient, cluster_configuration, prepared_controller_configuration: BaseNodesConfig
+    ):
+        """Destroy cluster via terraform"""
+
+        cluster_id = cluster_configuration.cluster_id
+        clusters = api_client.clusters_list() if not cluster_id else [{"id": cluster_id}]
+
+        for cluster_info in clusters:
+            cluster = Cluster(api_client, ClusterConfig(cluster_id=cluster_info["id"]), InfraEnvConfig())
+            controller = None
+            if cluster._config.platform == consts.Platforms.NUTANIX:
+                controller = NutanixController(prepared_controller_configuration, cluster_configuration)
+            if cluster._config.platform == consts.Platforms.VSPHERE:
+                controller = VSphereController(prepared_controller_configuration, cluster_configuration)
```

Comment:
Maybe create a method under `BaseTest` class that returns the controller by the `controller_configuration` and `cluster_configuration`

Reply:
Return it here also https://github.com/openshift/assisted-test-infra/blob/master/src/tests/base_test.py#L240-#L246

Reply:
Right, we have `NodeController` fixture